iocage/lib/Jail.py

```diff
@@ -372,6 +372,89 @@ def start(
 
         self._run_hook("poststart")
 
+    def fork_exec(
+        self,
+        command: typing.List[str],
+        error_handler: typing.Optional[typing.Callable[
+            [typing.Tuple[subprocess.Popen, str, str]],
+            typing.Tuple[bool, str],
+        ]]=None,
+        **temporary_config_override
+    ) -> typing.Generator['iocage.lib.events.IocageEvent', None, None]:
+        """
+        Starts a jail, runs a command and exists
+        """
+        self.require_jail_existing()
+        self.require_jail_stopped()
+
+        events: typing.Any = iocage.lib.events
+        jailForkExecEvent = events.JailForkExec(jail=self)
+        jailExecEvent = events.JailExec(jail=self)
+
+        yield jailForkExecEvent.begin()
+
+        original_config = self.config
+        config_data = original_config.data
+
+        for key, value in temporary_config_override.items():
+            config_data[key] = value
+
+        self.config = iocage.lib.Config.Jail.JailConfig.JailConfig(
+            data=original_config.data,
+            host=self.host,
+            jail=self,
+            logger=self.logger
+        )
+
+        for event in self.start():
+            yield event
+
+        def _revert_jail_started():
+            for event in self.stop(force=True):
+                yield event
+        jailExecEvent.add_rollback_step(_revert_jail_started)
+
+        yield jailExecEvent.begin()
+
+        child, stdout, stderr = self.exec(command, ignore_error=True)
+        if child.returncode != 0:
+            if (error_handler is None):
+                passed = False
+                message = ""
+            else:
+                passed, message = error_handler(child, stdout, stderr)
+
+            if passed is False:
+                err = iocage.lib.errors.CommandFailure(
+                    returncode=child.returncode
+                )
+                yield jailExecEvent.fail(err)
+                raise err
+            else:
+                yield jailExecEvent.skip(err)
+                self.logger.debug(message)
+        else:
+            yield jailExecEvent.end()
```

Comment:
does `yield` leave this function? (i.e.: can we swap the `if`/`else` so that we unindent the entire current `if`-block.